File: camerax_android/lib/src/legacy/view/use_case.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension UseCaseArgs on $interface.UseCase {
  $native.UseCase get obj {
    switch (this) {
      case $interface.UseCase.imageCapture:
        return $native.UseCase.imageCapture;
      case $interface.UseCase.imageAnalysis:
        return $native.UseCase.imageAnalysis;
      case $interface.UseCase.videoCapture:
        return $native.UseCase.videoCapture;
    }
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/view/UseCase.kt
package dev.hebei.camerax_android.legacy.view

import dev.hebei.camerax_android.legacy.UseCase

val UseCase.obj
    get() = when (this) {
        UseCase.IMAGE_CAPTURE -> androidx.camera.view.CameraController.IMAGE_CAPTURE
        UseCase.IMAGE_ANALYSIS -> androidx.camera.view.CameraController.IMAGE_ANALYSIS
        UseCase.VIDEO_CAPTURE -> androidx.camera.view.CameraController.VIDEO_CAPTURE
    }